.github/workflows/elixir.yml

```diff
@@ -1,27 +1,89 @@
 name: TableRex CI
 
-on:
-  push:
-    branches: [master]
-  pull_request:
-    branches: [master]
+on: push
 
 jobs:
-  build:
+  # Official support is for latest 3 minor Elixir
+  # versions & latest 2 major OTP releases.
+  elixir_1_15:
     runs-on: ubuntu-latest
 
+    name: OTP ${{matrix.otp}} / Elixir ${{matrix.elixir}}
+
+    strategy:
+      matrix:
+        otp: [24.x, 25.x, 26.x]
+        elixir: [1.15.x]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Elixir
+        uses: erlef/setup-beam@v1
+        with:
+          elixir-version: ${{ matrix.elixir }}
+          otp-version: ${{ matrix.otp }}
+
+      - name: Install dependencies
+        run: mix deps.get
+
+      - name: Compile & error on warning
+        run: mix compile --warnings-as-errors
+
+      - name: Check code is formatted
+        run: mix format --check-formatted
+
+      - name: Run the test suite
+        run: mix test
+
+
+  elixir_1_14:
+    runs-on: ubuntu-latest
+
+    name: OTP ${{matrix.otp}} / Elixir ${{matrix.elixir}}
+
+    strategy:
+      matrix:
+        otp: [24.x, 25.x]
+        elixir: [1.14.x]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Elixir
+        uses: erlef/setup-beam@v1
+        with:
+          elixir-version: ${{ matrix.elixir }}
+          otp-version: ${{ matrix.otp }}
+
+      - name: Install dependencies
+        run: mix deps.get
+
+      - name: Compile & error on warning
+        run: mix compile --warnings-as-errors
+
+      - name: Check code is formatted
+        run: mix format --check-formatted
+
+      - name: Run the test suite
+        run: mix test
+
+
+  elixir_1_13:
+    runs-on: ubuntu-latest
+
+    name: OTP ${{matrix.otp}} / Elixir ${{matrix.elixir}}
+
     strategy:
       matrix:
-        # Official support is for latest 3 minor Elixir
-        # versions & latest 2 major OTP releases.
-        elixir: ["1.8", "1.9", "1.10"]
-        otp: ["21", "22"]
```

Comment:
First off, thanks for the change! Definitely needed.

I think the GitHub Action workflow has been split out into 3 jobs because Elixir 1.13 & 1.14 run against Erlang OTP 24 & 25 - but Elixir 1.15 also runs against Erlang OTP 26 – is that correct?

If so, we can use the `[exclude](https://stackoverflow.com/a/68994907/1774000)` function here as part of the matrix definition to exclude specific combos that we know should not run. That way we can stick to just having the one Job definition and let the matrix handle the rest.

Reply:
Sure no problem - updated to consolidate again